### ISSUE TITLE
Components types for Vite.js projects

### ISSUE THREAD

**Issue by eduardo-wrkt** [mondaycom/monday-ui-react-core#1598]
Hi there!

First of all I would like to start by saying that I love this component library, it's simple, clean and beautiful ❤️.

I usually create my Monday apps using Vite.js and always face the same issue, the component types.

The main problem is that in the **package.json** file of this project the types are not exported, so when I try importing the components in my project I get a Typescript error and also don't have access to the component's props.

This is the error that I get when trying to import a component into my project:
`Could not find a declaration file for module 'monday-ui-react-core'. 'component_path.js' implicitly has an 'any' type.`
`There are types at 'types_file_path.d.ts', but this result could not be resolved when respecting package.json "exports". The 'monday-ui-react-core' library may need to update its package.json or typings.`

I've been searching in the closed issues of this repository and found this [issue](https://github.com/mondaycom/monday-ui-react-core/issues/1423) that reports the same problem.

Apparently this was fixed in the release **2.17.2**, but for some reason, the current version (I'm using the version **2.38.0**) still has this problem.

Can you guys tell me why this happened? 🤔

It's a simple fix, you guys only need to change some lines in the package.json file:

## Current package.json
```
...
"exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/esm/src/index.js",
      "default": "./dist/main.js"
    },
    "./dist/": "./dist/",
    "./icons": {
      "import": "./dist/esm/src/components/Icon/Icons/icons.js",
      "default": "./dist/allIcons.js"
    },
    "./interactionsTests": "./dist/esm/src/tests/interactionsTests.js",
    "./testIds": "./dist/esm/src/tests/testIds.js",
    "./storybookComponents": "./dist/storybook/index.js",
    "./next": "./dist/esm/src/next/next.js",
    "./tokens": "./dist/tokens/tokens.css",
    "./mockedClassNames": {
      "import": "./dist/mocked_classnames_esm/src/index.js"
    }
  },
...
```

## Fixed package.json
```
...
"exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/esm/src/index.js",
      "default": "./dist/main.js",
      "types": "./dist/types.d.ts"
    },
    "./dist/": "./dist/",
    "./icons": {
      "import": "./dist/esm/src/components/Icon/Icons/icons.js",
      "default": "./dist/esm/src/components/Icon/Icons/icons.js",
      "types": "./dist/esm/components/Icon/Icons/index.d.ts"
    },
    "./interactionsTests": "./dist/esm/src/tests/interactionsTests.js",
    "./testIds": "./dist/esm/src/tests/testIds.js",
    "./storybookComponents": "./dist/storybook/index.js",
    "./next": "./dist/esm/src/next/next.js",
    "./tokens": "./dist/tokens/tokens.css",
    "./mockedClassNames": {
      "import": "./dist/mocked_classnames_esm/src/index.js"
    }
  },
...
```

Thanks in advance!

**Comment by SergeyRoyt** [comment]
Hey, Eduardo!
First of all, thanks for using us and for the feedback! ❤️

About the Vite related issues and why they've returned - we've indeed released a fix you are proposing in the version `2.17.2`, however we quickly found out that these changes affect the webpack-build of our other internal library, which is a consumer of `monday-ui-react-core`. So in the version `2.18.1` we've reverted the changes and luckily forgot to update about that in the corresponding issue, sorry :(
I'll try to look into other solutions to make it work for all the bundlers during next 2 weeks - will update here 🙏

**Comment by SergeyRoyt** [comment]
@eduardo-wrkt hey!
It took a bit longer then expected, but the Vite fix is released in version `2.56.0` 🥳
This one we tested better, so we don't expect any issues with it. But anyway we'd appreciate if you double-check if it works well on your side 
Thank you!

**Comment by eduardo-wrkt** [comment]
Hey @SergeyRoyt!

I've been testing it for a couple of days and it seems to be working perfectly 😄.

Thank you for your help!
